src/LEGO.AsyncAPI.Readers/V2/AsyncApiMessageDeserializer.cs

```diff
@@ -19,19 +19,19 @@ internal static partial class AsyncApiV2Deserializer
         private static readonly FixedFieldMap<AsyncApiMessage> messageFixedFields = new()
         {
             {
-                "messageId", (a, n) => { a.MessageId = n.GetScalarValue(); }
+                "messageId", (a, n) => { }
             },
             {
-                "headers", (a, n) => { a.Headers = AsyncApiSchemaDeserializer.LoadSchema(n); }
+                "headers", (a, n) => { a.Headers = new AsyncApiMultiFormatSchema { Schema = AsyncApiSchemaDeserializer.LoadSchema(n) }; }
             },
             {
-                "payload", (a, n) => { a.Payload = null; /* resolved after the initial run */ }
+                "payload", (a, n) => { a.Payload = null; }
             },
             {
                 "correlationId", (a, n) => { a.CorrelationId = LoadCorrelationId(n); }
             },
             {
-                "schemaFormat", (a, n) => { a.SchemaFormat = LoadSchemaFormat(n.GetScalarValue()); }
+                "schemaFormat", (a, n) => { a.Headers.SchemaFormat = n.GetScalarValue(); }
```

Comment:
when using the multiformatschema deserializer, this can be skipped. (simply adding a comment `/* loaded as part of multiformatschema */` would probably be best here.

Reply:
This references the Payload schemaformat, not the header schemaformat.